### ISSUE TITLE
Each record should implement a universal record type

### ISSUE THREAD

**Issue by tobicrain** [patmood/pocketbase-typegen#9]

**Comment by benallfree** [comment]
@tobicrain Agreed, I use something like:

```ts
export type RecordId = string
export type UserId = RecordId
export type IsoDate = string

export const pocketNow = () => new Date().toISOString()

export type RecordBase = {
   id: RecordId
   created: IsoDate
   updated: IsoDate
}
```

**Comment by patmood** [comment]
Does a base type have any practical use, given that all the types are generated? 

I can see why you would want to do that in a library where you're writing types manually. Aliases seem like a good idea though. Will track that in #12

**Comment by benallfree** [comment]
Yeah you might have a point about the base type. I'm open to either approach :)

**Comment by patmood** [comment]
I didn't realise the "base" fields (id/created/updated)  were missing from the schema. I can add this in but when PB adds support for generics, those fields may already be included.

**Comment by benallfree** [comment]
Agreed probably best to wait for 0.8 or ask

**Comment by tobicrain** [comment]
My Record Interface looked something like this:

export interface Record {
  id?: string;
  created?: string;
  updated?: string;
  '@collectionId'?: string;
  '@collectionName'?: string;
  '@expand'?: Expand;
}
export interface Expand {}

**Comment by patmood** [comment]
Hey @tobicrain in `1.0.11` I added
```
export type BaseRecord = {
    id: RecordIdString
    created: IsoDateString
    updated: IsoDateString
    "@collectionId": string
    "@collectionName": string
}
```
But I didn't made every record extend it yet. When 0.8 is closer I'll make sure it works great.

**Comment by timohermans** [comment]
Was looking for this, nice that it's on the radar. Love this package @patmood , good job <3

**Comment by patmood** [comment]
It looks like PocketBase 0.8 will not include the base types, so they should be included in this package.

I just published v1.0.12 which adds Response types, eg ProfilesResponse. This is an intersection type with BaseRecord and has fields like id, created, updated etc.

This means the original ProfilesRecord type can still be used when creating records.

PocketBase 0.8 has quite a few other changes, so I'll need to make make a breaking change to support it in the future.
